src/com/google/cloud/datastore/types.js

```diff
@@ -262,9 +260,9 @@ foam.CLASS({
 
       var name = new Array(ids.length);
```

Comment:
I think this is meant to be `name = ''`, and use `+=` on the string to avoid creating garbage with arrays.